.github/workflows/python.yml

```diff
@@ -1,62 +1,125 @@
-name: Python CI Workflow
+name: Python CI/CD
 
 on:
   push:
   pull_request:
+  release:
+    types:
+      - published
 
 jobs:
 
-  build:
-    name: '[Python ${{ matrix.python-version }}]'
-    runs-on: ubuntu-20.04
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version:
-          - 3.6
           - 3.8
 
     steps:
 
       - uses: actions/checkout@master
       - run: git fetch --prune --unshallow
 
-      - name: Set up Python
+      - name: Install Python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Install Python tools
-        run: pip install setuptools_scm
+      - name: Install sdist dependencies
+        run: pip install setuptools_scm cmake_build_extension
+
+      - name: Build sdist
+        run: python setup.py sdist
 
       - name: Install dependencies
         run: |
           sudo apt-get update
           sudo apt-get install -y \
-              build-essential coinor-libipopt-dev libeigen3-dev swig \
-              libxml2-dev libassimp-dev
-
-      - name: Create sdist
-        run: python setup.py sdist
+            build-essential \
+            coinor-libipopt-dev \
+            libeigen3-dev swig \
+            libxml2-dev \
+            libassimp-dev
 
       - name: Install sdist
         run: pip -v install dist/*.tar.gz
 
       - name: Test import
         run: python -c 'import idyntree.bindings'
 
-      - name: Remove sdist
-        run: |
-          rm -rf dist/
-          pip uninstall -y idyntree
+      - uses: actions/upload-artifact@v2
+        if: github.repository == 'robotology/idyntree'
+        with:
+          path: dist/*.tar.gz
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.8
+        os:
+          - ubuntu-latest
+          # - macos-latest
+          # - windows-latest
+
+    steps:
+
+      - uses: actions/checkout@master
+      - run: git fetch --prune --unshallow
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install cibuildwheel
+        run: pip install cibuildwheel
 
-      - name: Create bdist_wheel
-        run: pip wheel -v -w dist/ .
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+        env:
+          CIBW_BUILD_VERBOSITY: 1
+          CIBW_BUILD: cp38-manylinux_x86_64
+          CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
+          CIBW_BEFORE_BUILD: "yum install -y eigen3-devel assimp-devel libxml2-devel coin-or-Ipopt-devel swig3"
```

Comment:
What is the version of `eigen3` installed in this CentOS image? Old Eigen version have some quite subtle bugs, and for the future we may want to use some functionalities of newer Eigen versions. I don't think it is a problem to achieve this (we can easily add an option to use an internal Eigen via FetchContent, and enable it when building the wheels) but it is better to be aware of this.

Reply:
Good point, it seems to be `3.3.7`.

Reply:
How did you get this info?

Reply:
Checking the logs :)